src/LagrangianParticleTracking/update_particle_properties.jl

```diff
@@ -91,7 +91,7 @@ end
 @kernel function _advect_particles!(particles, restitution, grid::ImmersedBoundaryGrid, Δt, velocities)
     p = @index(Global)
 
-    old_pos = (particles.x[p], particles.y[p], particles.y[p])
+    old_pos = deepcopy((particles.x[p], particles.y[p], particles.z[p]))
```

Comment:
Do you need `deepcopy`?

Reply:
the positions change in `update_particle_positions!` just below. I am not sure if we need deepcopy though...

Reply:
One (unstated) requirement of the particle tracking algorithm is to support huge numbers of particles (hundreds of millions or billions) so we likely want to avoid the allocation a `deepcopy` would incur.